scripts/update_version.py

```diff
@@ -72,16 +76,21 @@ def update_version():
         file.write(replace_commit_id)
 
     changelog_line1 = "# [Unreleased](https://github.com/pybamm-team/PyBaMM/)\n"
-    changelog_line2 = f"# [v{release_version}](https://github.com/pybamm-team/PyBaMM/tree/v{release_version}) - {last_day_of_month}\n\n"
+    changelog_line2 = f"# [v{release_version}](https://github.com/pybamm-team/PyBaMM/tree/v{release_version}) - {release_date}\n\n"
 
     # CHANGELOG.md
     with open(os.path.join(pybamm.root_dir(), "CHANGELOG.md"), "r+") as file:
         output_list = file.readlines()
         output_list[0] = changelog_line1
+        # add a new heading for rc0 releases
         if "rc0" in release_version:
             output_list.insert(2, changelog_line2)
         else:
-            output_list[2] = changelog_line2
+            # for rcX and final releases, update the already existing rc
+            # release heading
+            for i in range(0, len(output_list)):
+                if re.search("[v]\d\d\.\drc\d", output_list[i]):
```

Comment:
For rc0 release, we just need to add a new heading under the "Unrelease" heading. But, for rcX and final releases, the workflow now finds the existing "rc" heading and replaces it with an updated version and date.